### PR TITLE
convert more to typescript

### PR DESCRIPTION
**Pull request by ndelangen**
Convert more components to typescript, so we get proper typings

<details>
  <summary>📦 Published PR as canary version: <code>5.6.9-canary.292.eb1b9b5.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/design-system@5.6.9-canary.292.eb1b9b5.0
  # or 
  yarn add @storybook/design-system@5.6.9-canary.292.eb1b9b5.0
  ```
</details>

### CODE REVIEW

**Comment by ndelangen** [comment]
@winkerVSbecks any idea why auto has decided to stop publishing? it worked last week, I think

**Comment by winkerVSbecks** [comment]
@ndelangen you need add a label: major, minor or patch

**Comment by winkerVSbecks** [comment]
@ndelangen Actually it's complaining about OTP https://github.com/storybookjs/design-system/runs/3400303114?check_suite_focus=true 

Did you change the settings for this package on npm?

File: src/components/tooltip/TooltipNote.tsx [hunk 1]
```diff
@@ -19,10 +18,13 @@ const Note = styled.div`
   margin: 6px;
 `;
 
-export function TooltipNote({ note, ...rest }) {
+export const TooltipNote = ({
+  note,
+  ...rest
+}: Props & Omit<ComponentProps<typeof Note>, 'children'> & { children?: never }) => {
```
Comment (winkerVSbecks) [inline src/components/tooltip/TooltipNote.tsx:24]: Nice! Wasn't aware of this `never` type

**Review by winkerVSbecks — APPROVED** [review]
Thanks for converting all these files!

**Comment by ndelangen** [comment]
I didn't change it, no..

**Pull request merged**